nuxeo-importer-core/src/main/java/org/nuxeo/ecm/platform/importer/base/GenericThreadedImportTask.java

```diff
@@ -360,19 +361,26 @@ public synchronized void run() {
             recursiveCreateDocumentFromNode(rootDoc, rootSource);
             session.save();
             GenericMultiThreadedImporter.addCreatedDoc(taskId, uploadedFiles);
-            TransactionHelper.commitOrRollbackTransaction();
+            completedAbruptly = false;
         } catch (Exception e) { // deals with interrupt below
             log.error("Error during import", e);
             ExceptionUtils.checkInterrupt(e);
             notifyImportError();
         } finally {
             log.info("End of task");
-            if (session != null) {
-                session.close();
-                session = null;
-            }
-            synchronized (this) {
-                isRunning = false;
+            try {
```

Comment:
Indeed that's a worthwhile refactoring but for this ticket I wanted the minimal fix possible.